[Timothy Wong] iP

timothywongej opened the pull request:
# Start lines with a '#' to create headings
## Multiple '##' in a row denote smaller heading sizes.
###### Up to six '#'. 

###### A bullet list
* Unordered List
* Item 2
    * Indented item

###### A numbered list
1. Ordered list 
2. Item 2
3. Item 3
    1. Indented item
    2. Indented item 2

###### No syntax highlighting
    if (indented with 4 spaces) {
        return true
    }

###### A fenced code block (with syntax highlighting)
```javascript
if (codeFencing) {
    return true
}
````

###### Tasklist
- [x] tasklist 
- [ ] incomplete item
- [x] complete item 

###### Emoji
:shipit: 

###### Quote
In the words of prof: 
> Unit and regression testing is important


###### Hyper link
https://github.com/nus-cs2103-AY2122S1/ip/pull/291 

###### Inline code
inline code blocks `var example = true`

###### Some text formatting
It's very easy to make some words **bold** and other words *italic* with Markdown.
You can even add ~~strikethroughs~~.

Commit:
Add Gradle support

timothywongej committed:
Ignore .class file structures

timothywongej committed:
Add Level-1 Greet, Echo, Exit

timothywongej committed:
Add Level-2, Add, List

timothywongej committed:
Add Level 3, Mark as Done

timothywongej committed:
Add Level-4, ToDos, Events, Deadlines

timothywongej committed:
Remove tab t in println

timothywongej committed:
Add A-TextUiTesting

timothywongej committed:
Add Level-5, Handle errors

timothywongej committed:
Add Level-6 Delete

timothywongej committed:
Add DukeException and exception handling

timothywongej committed:
Update A-TextUiTesting

timothywongej committed:
Add Level-7: Save, Write to Hard Disk

timothywongej committed:
Remove Event and Deadline string bug

timothywongej committed:
Add Level-8, Date

timothywongej committed:
Merge branch 'branch-Level-7'
Add Level-7: Save and writing to hard disk.

timothywongej committed:
Merge Level-8 onto master

timothywongej committed:
Add A-MoreOOP, restructure files

timothywongej committed:
Add A-Packages: duke package

timothywongej committed:
Add A-JUnit tests

timothywongej committed:
Create JAR File

timothywongej committed:
Add A-JavaDoc

timothywongej committed:
Update JUnit Test: Module & Dependencies

timothywongej committed:
Add A-JavaDoc: Header comments

timothywongej committed:
Add A-CodingStandard: Checked

timothywongej committed:
Add Level-9: Find

timothywongej committed:
Merge branch 'branch-A-JavaDoc'

timothywongej committed:
Merge branch 'branch-A-CodingStandard'

timothywongej committed:
Merge branch 'branch-Level-9'

timothywongej committed:
Update Task: make abstract

timothywongej committed:
Update README.md

timothywongej committed:
Merge branch 'master' of https://github.com/timothywongej/ip

timothywongej committed:
Change gradle version to 6.8.1

timothywongej committed:
Merge branch 'add-gradle-support'

timothywongej committed:
Update build.gradle: mainClassName

timothywongej committed:
Add checkstyle xml files

timothywongej committed:
Edit to checkstyle audit

timothywongej committed:
Add JavaFX dependencies

timothywongej committed:
Add Launcher class that drives GUI

timothywongej committed:
Add GUI iteration 1: Echo user

timothywongej committed:
Add GUI Iteration 2: Dialog Boxes

timothywongej committed:
Add GUI Iteration 3: Custom behavior

timothywongej committed:
Merge branch 'branch-Level-10' add GUI for duke.

sueann-chua commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,262 @@
+     * Converts input string to integer.
+     *
+     * @param req The full command
+     * @return Integer index of the task in the task list
```
Javadoc sentences should end with fullstops

sueann-chua commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Queries whether the Task is done.
+     * @return boolean isDone.
```
Should have empty line between description and return section

sueann-chua commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Returns the status icon depending on whether the Task is done.
+     * @return String Representation.
+     */
```
Again, should have empty line between description and return section

sueann-chua commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    /**
+     * Returns the description of the Task.
+     * @return Description.
```
Same as above, should have empty line

sueann-chua commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,262 @@
+     * Validates a find command.
+     *
+     * Receives a string to match.
+     * @param req
```
Could add description for parameter for consistency since your other method headers have param descriptions

sueann-chua reviewed (commented):
Overall, very few coding standard violations. Goodjob! :)

timothywongej committed:
Create Task package

timothywongej committed:
Move Task classes into Task package

timothywongej committed:
Refactor to include duke.task import

timothywongej committed:
Refactor to include duke.task import

timothywongej committed:
Remove main method in Duke

timothywongej committed:
Add Main class

timothywongej committed:
Add FXML to Duke

timothywongej committed:
Refactor DialogBox for FXML

timothywongej committed:
Refactor Duke for FXML

timothywongej committed:
Change Launcher to launch Main.class

timothywongej committed:
Add Command package and Classes

timothywongej committed:
Add Command abstract class

timothywongej committed:
Add Task Commands

timothywongej committed:
Add additional command classes

timothywongej committed:
Update Parser: return Command objects

timothywongej committed:
Refactor Duke: support GUI and Command class

timothywongej committed:
Refactor Storage

timothywongej committed:
Refactor TaskList

timothywongej committed:
Refactor Ui

timothywongej committed:
Package DukeException

timothywongej committed:
Merge branch 'branch-Add-Commands'

Add Command class and refactored code to implement GUI.

timothywongej committed:
Add welcome message to GUI

The Duke GUI has to display a welcome message on startup.

MainWindow controller, Duke, Ui and Main classes interact to
retrieve and displasy Strings in the GUI.

Let's
* update Main: call method of controller
* update Duke, Ui and MainWindow to retrieve and display message

timothywongej committed:
Enable assertions

timothywongej committed:
Add assertions

Storage reads from the file and checks whether a task is
done (either 0 or 1).
DoneCommand and DeleteCommand have common behaviours
(take in an index).

Let's add assetions to check validity of indexes.

timothywongej committed:
Improve code quality

timothywongej committed:
Merge pull request #1 from timothywongej/branch-A-Assertions

Branch-A-Assertions

timothywongej committed:
Merge branch 'master' into branch-A-CodeQuality

master branch is ahead of branch-A-CodeQuality after PR merge with A-Assertions.

timothywongej committed:
Merge pull request #2 from timothywongej/branch-A-CodeQuality

Improve code quality

timothywongej committed:
Add B-Reminders

Deadlines have got a LocalDate field.

Let's
* enable reminders to prompt upcoming deadlines
* create a new command to remind the user

timothywongej committed:
Merge pull request #3 from timothywongej/branch-B-Reminders

Add B-Reminders

timothywongej committed:
Remove bug due to white space

timothywongej committed:
Remove redundant exception

timothywongej committed:
Handle more exceptions in DeadlineCommand

timothywongej committed:
Handle more exceptions in EventCommand

timothywongej committed:
Add exception handling

timothywongej committed:
Remove redundant imports

timothywongej committed:
Update Commands

timothywongej committed:
Add A-BetterGui

There is opportunity to customise the GUI.

The text label is not pleasant to look at.

The dispaly picture is occupying space.

Let's
* make the display picture a circle
* add background and borders to the text label

timothywongej committed:
Update code quality

timothywongej committed:
Add Platform exit and Timer to Exit Command

timothywongej committed:
Change rewrite file to append to file

timothywongej committed:
Update message success

timothywongej committed:
Add code credit comments

timothywongej committed:
Refactor Storage class

timothywongej committed:
Refactor Ui class

timothywongej committed:
Update code quality

timothywongej committed:
Add A-UserGuide and product screenshot

timothywongej committed:
Update String output values

timothywongej committed:
Update GUI stage title

timothywongej committed:
Update J-Unit Testing

timothywongej committed:
Update README.md

timothywongej committed:
Update readme

timothywongej committed:
Change images to reduce lag in GUI

timothywongej committed:
Update product screenshot

timothywongej committed:
Refactor exit command to fix bug
